src/server/journal/journal_slice.cc

```diff
@@ -114,6 +114,25 @@ error_code JournalSlice::Close() {
 
 void JournalSlice::AddLogRecord(const Entry& entry, bool await) {
   DCHECK(ring_buffer_);
+
+  if (entry.opcode != Op::NOOP) {
```

Comment:
why did you need to move this above?

Reply:
It prevents a pretty benign race - so a change can't be visible in a replica but the LSN you'd read in the master would still be the old one.

If you have multiple replicas that's more of an issue, but still conceptually I think it makes sense to put it in the journal before it's replicated